lucene/core/src/java/org/apache/lucene/search/IndriAndWeight.java

```diff
@@ -38,8 +38,21 @@ public IndriAndWeight(
     this.boost = boost;
     this.scoreMode = scoreMode;
     weights = new ArrayList<>();
+    // Calculate total boost score so that boost can be normalized
+    float boostSum = 0;
     for (BooleanClause c : query) {
-      Weight w = searcher.createWeight(c.getQuery(), scoreMode, 1.0f);
+      if (c.getQuery() instanceof BoostQuery) {
+        boostSum += ((BoostQuery) c.getQuery()).getBoost();
```

Comment:
subjective: perhaps introduce a local variable to avoid duplicate `c.getQuery()` calls.

```suggestion
      Query q = c.getQuery();
      if (q instanceof BoostQuery) {
        boostSum += ((BoostQuery) q).getBoost();
```